packages/flutter/test/rendering/layers_test.dart

```diff
@@ -239,4 +290,169 @@ void main() {
       layer.shadowColor = const Color(1);
     });
   });
+
+  group('PhysicalModelLayer checks elevations', () {
+    /// Adds the layers to a container where A paints before B.
+    ///
+    /// Expects there to be `errorCount` errors.  Checking elevations is enabled
+    /// by default.
+    void _testConflicts(
+      PhysicalModelLayer layerA,
+      PhysicalModelLayer layerB,
+      int errorCount, [
+      bool enableCheck = true,
+    ]) {
+      assert(enableCheck || errorCount == 0, 'Cannot disable check and expect non-zero error count.');
+      final OffsetLayer container = OffsetLayer();
+      container.append(layerA);
+      container.append(layerB);
+      debugCheckElevationsEnabled = enableCheck;
+      debugDisableShadows = false;
+      int errors = 0;
+      if (enableCheck) {
+        FlutterError.onError = (FlutterErrorDetails details) {
+          errors++;
+        };
+      }
+      container.buildScene(SceneBuilder());
+      expect(errors, errorCount);
+      debugCheckElevationsEnabled = false;
```

Comment:
Why this off-on switching?

Also, should this be restored to whatever it was before _testConflicts was called?

Reply:
We have to make sure we restore the debug flags between each test.

Reply:
Why set it to false and then true right after then? Wouldn't one of these be enough?

Reply:
Ahh I see, whups.  Removed the second one.